Sources/Sentry/SentrySerialization.m

```diff
@@ -107,6 +107,34 @@ + (NSData *_Nullable)dataWithEnvelope:(SentryEnvelope *)envelope
     return envelopeData;
 }
 
++ (NSString *)baggageEncodedDictionary:(NSDictionary *)dictionary
+{
+    NSMutableArray *items = [[NSMutableArray alloc] initWithCapacity:dictionary.count];
+
+    NSMutableCharacterSet *allowedSet = [NSCharacterSet.alphanumericCharacterSet mutableCopy];
+    [allowedSet addCharactersInString:@"-_"];
+    NSInteger currentSize = 0;
+
+    for (id key in dictionary.allKeys) {
+        id value = dictionary[key];
+        NSString *keyDescription =
+            [[key description] stringByAddingPercentEncodingWithAllowedCharacters:allowedSet];
+        NSString *valueDescription =
+            [[value description] stringByAddingPercentEncodingWithAllowedCharacters:allowedSet];
+
+        NSString *item = [NSString stringWithFormat:@"%@=%@", keyDescription, valueDescription];
+        if (item.length + currentSize <= 8192) {
```

Comment:
If this isn't the first item we might go up to 8193 here because of the `,` separator.

Reply:
After having a call this is wrong as the +1 also happens for the first item and thus fixes the problem.